engine/src/main/java/org/destinationsol/game/screens/ChangeShip.java

```diff
@@ -119,7 +120,8 @@ private void changeShip(SolGame game, SolShip hero, ShipItem selected) {
         SolShip newHero = game.getShipBuilder().build(game, hero.getPosition(), new Vector2(), hero.getAngle(), 0, hero.getPilot(),
                 hero.getItemContainer(), newConfig, newConfig.getMaxLife(), hull.getGun(false), g2, null,
                 ei, new ShipRepairer(), hero.getMoney(), hero.getTradeContainer(), hero.getShield(), hero.getArmor());
-        game.getObjMan().removeObjDelayed(hero);
+        game.getObjMan().removeObjDelayed(hero.getHero());
         game.getObjMan().addObjDelayed(newHero);
+        game.getHero().toSolShip(newHero); // HACK
```

Comment:
Helps to explain why, maybe convert into a `// TODO: This is a hack because X which we need to fix with by doing Y` :-)

Reply:
I don't actually even remember that comment or the reason for it. Guess it might be pretty useless, since I don't see anything that hack-ish in there